src/particleContainer/LinkedCells.cpp

```diff
@@ -966,8 +966,8 @@ RegionParticleIterator LinkedCells::getRegionParticleIterator(
 	return RegionParticleIterator(type, &_cells, offset, stride, startRegionCellIndex, regionDimensions, _cellsPerDimension, startRegion, endRegion);
 }
 
-void LinkedCells::deleteMolecule(Molecule &molecule, const bool& rebuildCaches) {
-	auto cellid = getCellIndexOfMolecule(&molecule);
+void LinkedCells::deleteMolecule(ParticleIterator &moleculeIter, const bool& rebuildCaches) {
```

Comment:
this function should boil down to
```c++
moleculeIter.deleteCurrentParticle();
```
if rebuildCaches is false!

Reply:
this might also fix the current segfault in one of the tests.

Reply:
done #2a1191d